sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -220,7 +220,21 @@ case class GpuElementAt(left: Expression, right: Expression, failOnError: Boolea
           }
         }
       case _: MapType =>
-        (_, _) => throw new UnsupportedOperationException("non-literal key is not supported")
+        (map, indices) => {
+          if (failOnError) {
+            withResource(map.getMapKeyExistence(indices)) { keyExists =>
```

Comment:
in a conventional programming checking for key existence before doing the value lookup is considered a peroformance antipattern leading to an unnecessary second lookup. 

Is there a way to just build a logic with a single getMapValue?

Reply:
I thought about doing it without an extra lookup but there is no way for us to know if the map actually had a `null` value or was the key not found in the map.

Reply:
This is very easy to achieve by re-implementing map lookup to return a pair of `{value, success}`. Otherwise, checking for existence then retrieving is doubly expensive.

Reply:
I would prefer to go back and do that rather than accepting this double computation.